crates/router/src/connector/worldline.rs

```diff
@@ -651,27 +653,130 @@ impl ConnectorIntegration<api::RSync, types::RefundsData, types::RefundsResponse
     }
 }
 
+fn is_endpoint_verification(headers: &actix_web::http::header::HeaderMap) -> bool {
+    headers
+        .get("x-gcs-webhooks-endpoint-verification")
+        .is_some()
+}
+
 #[async_trait::async_trait]
 impl api::IncomingWebhook for Worldline {
-    fn get_webhook_object_reference_id(
+    fn get_webhook_source_verification_algorithm(
         &self,
         _request: &api::IncomingWebhookRequestDetails<'_>,
+    ) -> CustomResult<Box<dyn crypto::VerifySignature + Send>, errors::ConnectorError> {
+        Ok(Box::new(crypto::HmacSha256))
+    }
+
+    fn get_webhook_source_verification_signature(
+        &self,
+        request: &api::IncomingWebhookRequestDetails<'_>,
+    ) -> CustomResult<Vec<u8>, errors::ConnectorError> {
+        let header_value = request
+            .headers
+            .get("X-GCS-Signature")
+            .ok_or(errors::ConnectorError::WebhookSignatureNotFound)?
+            .to_str()
+            .into_report()
+            .change_context(errors::ConnectorError::WebhookSignatureNotFound)?;
+
+        let signature = consts::BASE64_ENGINE
+            .decode(header_value.as_bytes())
+            .into_report()
+            .change_context(errors::ConnectorError::WebhookSourceVerificationFailed)?;
```

Comment:
ideally we should only pass the actual signature, in `verify signature` we need to encode the signature we created. this will help in avoiding the unnecessary decoding in all the places.

Reply:
By default signature(binary data) will be encoded with encoding algorithm (like base64, base32, hex, decimal, base85), to get an actual signature we need to decode using respective algorithm. This encoding logic differs by connectors, but whatever the encoding algorithm is the signature will be useful only after decoding.